[dynamo] Mark a vt unspecialized nn module variable source earlier

anijain2305 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #155083
* #154867
* __->__ #154780

I am working on providing some skip guard helper functions to allow users to reduce guard overhead. This is a refactor to allow that.

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @kadeng @chauhang @amjames

anijain2305 committed:
[dynamo] Mark a vt unspecialized nn module variable source earlier

[ghstack-poisoned]

anijain2305 committed:
Update on "[dynamo] Mark a vt unspecialized nn module variable source earlier"


I am working on providing some skip guard helper functions to allow users to reduce guard overhead. This is a refactor to allow that.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames

[ghstack-poisoned]

anijain2305 committed:
Update on "[dynamo] Mark a vt unspecialized nn module variable source earlier"


I am working on providing some skip guard helper functions to allow users to reduce guard overhead. This is a refactor to allow that.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames

[ghstack-poisoned]

anijain2305 committed:
Update on "[dynamo] Mark a vt unspecialized nn module variable source earlier"


I am working on providing some skip guard helper functions to allow users to reduce guard overhead. This is a refactor to allow that.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames

[ghstack-poisoned]

anijain2305 committed:
Update on "[dynamo] Mark a vt unspecialized nn module variable source earlier"


I am working on providing some skip guard helper functions to allow users to reduce guard overhead. This is a refactor to allow that.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames

[ghstack-poisoned]

anijain2305 commented on torch/_dynamo/variables/builder.py:
```diff
@@ -1749,12 +1753,23 @@ def wrap_module(self, value: torch.nn.Module):
-                result = UnspecializedBuiltinNNModuleVariable(value, source=self.source)
+            if (
+                value.__module__.startswith(("torch.nn.modules", "torch.ao."))
+                and not value.__module__.startswith("torch.nn.modules.container")
```
`container` is deliberately left out to be bulitin because a container, like Sequential, can consist of user defined module.

anijain2305 committed:
Update on "[dynamo] Mark a vt unspecialized nn module variable source earlier"


I am working on providing some skip guard helper functions to allow users to reduce guard overhead. This is a refactor to allow that.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames

[ghstack-poisoned]

anijain2305 committed:
Update on "[dynamo] Mark a vt unspecialized nn module variable source earlier"


I am working on providing some skip guard helper functions to allow users to reduce guard overhead. This is a refactor to allow that.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames

[ghstack-poisoned]

anijain2305 committed:
Update on "[dynamo] Mark a vt unspecialized nn module variable source earlier"


I am working on providing some skip guard helper functions to allow users to reduce guard overhead. This is a refactor to allow that.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames

[ghstack-poisoned]

anijain2305 committed:
Update on "[dynamo] Mark a vt unspecialized nn module variable source earlier"


I am working on providing some skip guard helper functions to allow users to reduce guard overhead. This is a refactor to allow that.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames

[ghstack-poisoned]

anijain2305 committed:
Update on "[dynamo] Mark a vt unspecialized nn module variable source earlier"


I am working on providing some skip guard helper functions to allow users to reduce guard overhead. This is a refactor to allow that.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames

[ghstack-poisoned]

anijain2305 committed:
Update on "[dynamo] Mark a vt unspecialized nn module variable source earlier"


I am working on providing some skip guard helper functions to allow users to reduce guard overhead. This is a refactor to allow that.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames

[ghstack-poisoned]

anijain2305 committed:
Update on "[dynamo] Mark a vt unspecialized nn module variable source earlier"


I am working on providing some skip guard helper functions to allow users to reduce guard overhead. This is a refactor to allow that.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames

[ghstack-poisoned]

anijain2305 committed:
Update on "[dynamo] Mark a vt unspecialized nn module variable source earlier"


I am working on providing some skip guard helper functions to allow users to reduce guard overhead. This is a refactor to allow that.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames

[ghstack-poisoned]

StrongerXi commented on benchmarks/dynamo/ci_expected_accuracy/cpu_inductor_torchbench_inference.csv:
```diff
@@ -138,7 +138,7 @@ hf_Bert_large,pass,0
 
 
-hf_BigBird,pass,18
+hf_BigBird,pass,24
```
Is this an increase in # of graph breaks?

StrongerXi commented on torch/_dynamo/variables/builder.py:
```diff
@@ -1749,12 +1753,23 @@ def wrap_module(self, value: torch.nn.Module):
+            ) or getattr(value.__class__, "_dynamo_marked_static", False):
+                new_source = self.source
+                if config.inline_inbuilt_nn_modules:
+                    # Export corner case - look at test_repros.py test_inlining_cornercase
+                    new_source = UnspecializedBuiltinNNModuleSource(self.source)
```
Mind dropping link to #147030 here and below? I plan to take a stab at it.

anijain2305 replied:
@Lucaskabela has been looking at it

StrongerXi replied:
Oh sweet!

StrongerXi commented on torch/_dynamo/variables/builder.py:
```diff
@@ -434,7 +436,10 @@ def __call__(self, value):
 
         vt = self._wrap(value)
-        vt.source = self.source
+
+        if vt.source is None:
+            vt.source = self.source
```
I thought `self._wrap` will properly assign the source already, sad...

williamwen42 reviewed (commented):
Is the increased number of graph breaks acceptable?

anijain2305 committed:
Update on "[dynamo] Mark a vt unspecialized nn module variable source earlier"


I am working on providing some skip guard helper functions to allow users to reduce guard overhead. This is a refactor to allow that.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames

[ghstack-poisoned]

anijain2305 committed:
Update on "[dynamo] Mark a vt unspecialized nn module variable source earlier"


I am working on providing some skip guard helper functions to allow users to reduce guard overhead. This is a refactor to allow that.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames

[ghstack-poisoned]

seemethere commented:
@pytorchbot revert -c ghfirst -m "This fails internal testing see, https://fburl.com/diff/b0yuxk4w"

anijain2305 commented:
Relanding at https://github.com/pytorch/pytorch/pull/155099
